lib/src/over_react_test/common_component_util.dart

```diff
@@ -223,6 +231,11 @@ void _testPropForwarding(BuilderOnlyUiFactory factory, dynamic childrenFactory()
       ...nonDefaultForwardingTestProps,
     };
 
+    /// The props we expect to be present on the `forwardingTarget`.
+    final unconsumedProps = <String, dynamic>{
```

Comment:
Shouldn't this go below  
```
unconsumedPropKeys.forEach(propsThatShouldNotGetForwarded.remove);
```
?
Otherwise that doc comment isn't really accurate

Also, isnc't this already handled by ` ...defaultProps,` in `propsThatShouldNotGetForwarded`?

Reply:
I mean... its been a month+... but I definitely tried putting this in both places, and had false negative failures in WSD when `propsThatShouldNotGetForwarded` keys were not removed. IIRC, it had something to do with `getSkippedPropKeys` or `nonDefaultForwardingTestProps`.

Reply:
My question still stands here; for which case is this needed?

Reply:
Three tests fail if this is removed:

1. <details><summary>commonComponentTests should pass when the correct unconsumed props are specified forwards unconsumed props as expected</summary>

    ```00:00 +6 -1: commonComponentTests should pass when the correct unconsumed props are specified forwards unconsumed props as expected [E]
      Expected: empty
        Actual: WhereIterable<dynamic>:[
                  'ShouldBeForwardedProps.foo',
                  'ShouldBeForwardedProps.foo2'
                ]
      UnconsumedProps were not forwarded.

      These prop keys were not found within the props of the forwarding target:

          ShouldBeForwardedProps.foo,
          ShouldBeForwardedProps.foo2

      But they were expected to be there since they live within one of the prop mixins specified
      in the list returned to `commonComponentTests.getUnconsumedPropKeys()` in your test.

      If these props should be forwarded to the forwarding target rendered by this component,
      the value of `UiComponent2.consumedProps` must be overridden to exclude those keys:

          // Option 1: forward everything
          @override
          get consumedProps => [];

          // Option 2: specify the prop mixins that should be consumed:
          @override
          get consumedProps => propsMeta.forMixins({
            SomeOtherPropsMixin,
            // leave out the mixin(s) containing ShouldBeForwardedProps.foo, ShouldBeForwardedProps.foo2
          });

          // Option 3: specify the prop mixins that should NOT be consumed:
          @override
          get consumedProps => propsMeta.allExceptForMixins({
            // list the mixin(s) containing ShouldBeForwardedProps.foo, ShouldBeForwardedProps.foo2
          });

      If these props should NOT be forwarded to the forwarding target rendered by this component,
      remove these lines from the list returned to `commonComponentTests.getUnconsumedPropKeys()`:

          propsMeta.forMixin(/* name of the mixin containing ShouldBeForwardedProps.foo */).keys,
          propsMeta.forMixin(/* name of the mixin containing ShouldBeForwardedProps.foo2 */).keys

      package:test_api                                                              expect$
      package:over_react_test/src/over_react_test/common_component_util.dart 280:9  <fn>
    ```
</details>

2. <details><summary>commonComponentTests should skip checking for certain props forwards unconsumed props as expected</summary>

    ```00:00 +17 -2: commonComponentTests should skip checking for certain props forwards unconsumed props as expected [E]
      Expected: empty
        Actual: WhereIterable<dynamic>:[
                  'ShouldBeForwardedProps.foo',
                  'ShouldBeForwardedProps.foo2'
                ]
      UnconsumedProps were not forwarded.

      These prop keys were not found within the props of the forwarding target:

          ShouldBeForwardedProps.foo,
          ShouldBeForwardedProps.foo2

      But they were expected to be there since they live within one of the prop mixins specified
      in the list returned to `commonComponentTests.getUnconsumedPropKeys()` in your test.

      If these props should be forwarded to the forwarding target rendered by this component,
      the value of `UiComponent2.consumedProps` must be overridden to exclude those keys:

          // Option 1: forward everything
          @override
          get consumedProps => [];

          // Option 2: specify the prop mixins that should be consumed:
          @override
          get consumedProps => propsMeta.forMixins({
            SomeOtherPropsMixin,
            // leave out the mixin(s) containing ShouldBeForwardedProps.foo, ShouldBeForwardedProps.foo2
          });

          // Option 3: specify the prop mixins that should NOT be consumed:
          @override
          get consumedProps => propsMeta.allExceptForMixins({
            // list the mixin(s) containing ShouldBeForwardedProps.foo, ShouldBeForwardedProps.foo2
          });

      If these props should NOT be forwarded to the forwarding target rendered by this component,
      remove these lines from the list returned to `commonComponentTests.getUnconsumedPropKeys()`:

          propsMeta.forMixin(/* name of the mixin containing ShouldBeForwardedProps.foo */).keys,
          propsMeta.forMixin(/* name of the mixin containing ShouldBeForwardedProps.foo2 */).keys

      package:test_api                                                              expect$
      package:over_react_test/src/over_react_test/common_component_util.dart 280:9  <fn>
    ```
</details>

3. <details><summary>commonComponentTests does not call `factory` directly within the consuming group when passed a UiComponent2 (new boilerplate) forwards unconsumed props as expected</summary>

    ```00:00 +38 -3: commonComponentTests does not call `factory` directly within the consuming group when passed a UiComponent2 (new boilerplate) forwards unconsumed props as expected [E]
      Expected: empty
        Actual: WhereIterable<dynamic>:[
                  'ShouldBeForwardedProps.foo',
                  'ShouldBeForwardedProps.foo2'
                ]
      UnconsumedProps were not forwarded.

      These prop keys were not found within the props of the forwarding target:

          ShouldBeForwardedProps.foo,
          ShouldBeForwardedProps.foo2

      But they were expected to be there since they live within one of the prop mixins specified
      in the list returned to `commonComponentTests.getUnconsumedPropKeys()` in your test.

      If these props should be forwarded to the forwarding target rendered by this component,
      the value of `UiComponent2.consumedProps` must be overridden to exclude those keys:

          // Option 1: forward everything
          @override
          get consumedProps => [];

          // Option 2: specify the prop mixins that should be consumed:
          @override
          get consumedProps => propsMeta.forMixins({
            SomeOtherPropsMixin,
            // leave out the mixin(s) containing ShouldBeForwardedProps.foo, ShouldBeForwardedProps.foo2
          });

          // Option 3: specify the prop mixins that should NOT be consumed:
          @override
          get consumedProps => propsMeta.allExceptForMixins({
            // list the mixin(s) containing ShouldBeForwardedProps.foo, ShouldBeForwardedProps.foo2
          });

      If these props should NOT be forwarded to the forwarding target rendered by this component,
      remove these lines from the list returned to `commonComponentTests.getUnconsumedPropKeys()`:

          propsMeta.forMixin(/* name of the mixin containing ShouldBeForwardedProps.foo */).keys,
          propsMeta.forMixin(/* name of the mixin containing ShouldBeForwardedProps.foo2 */).keys

      package:test_api                                                              expect$
      package:over_react_test/src/over_react_test/common_component_util.dart 280:9  <fn>
    ```

</details>

Reply:
I was more trying to get at: which case is this addition and related expectation needed for to make tests fail as expected?

Those don't fail for me if I remove the expectations that rely on unconsumedProps being in place:
```
expect(missingUnconsumedPropKeys, isEmpty,
```

However, this one does:
> commonComponentTests fails when a component does not forward all of the unconsumed props in propsMeta - (tests fail check)

So that's really the reason this code is here, and I'm satisfied by that answer!